src/theory/datatypes/sygus_datatype_utils.cpp

```diff
@@ -605,6 +606,40 @@ Node getExpandedDefinitionForm(Node op)
   return eop.isNull() ? op : eop;
 }
 
+void computeExpandedDefinitionForms(Env& env, const TypeNode& tn)
+{
+  std::unordered_set<TypeNode> processed;
+  std::vector<TypeNode> toProcess;
+  toProcess.push_back(tn);
+  size_t index = 0;
+  while (index < toProcess.size())
+  {
+    TypeNode tnp = toProcess[index];
+    index++;
+    Assert(tnp.isSygusDatatype());
+    const DType& dt = tnp.getDType();
+    const std::vector<std::shared_ptr<DTypeConstructor>>& cons =
+        dt.getConstructors();
+    for (const std::shared_ptr<DTypeConstructor>& c : cons)
```

Comment:
I checked [cppreference](https://en.cppreference.com/w/cpp/language/range-for#Temporary_range_expression) and I believe you can trust the compiler to handle temporaries correctly.
```suggestion
    for (const std::shared_ptr<DTypeConstructor>& c : dt.getConstructors())
```

Reply:
I don't like this change since it makes it seem as though `dt.getConstructors` is recomputed every iteration.